tests/test_networkmanager.py

```diff
@@ -514,6 +514,23 @@ def test_settings_secrets(self):
         # Secrets request should contain just vpn section with the secrets in
         self.assertEqual(connection_i.GetSecrets('vpn'), {'vpn': {'secrets': secrets}})
 
+    def test_get_conn_by_uuid(self):
+        uuid = '133d8eb9-6de6-444f-8b37-f40bf9e33226'
+        settings = dbus.Dictionary({
+            'connection': dbus.Dictionary({
+                'id': 'test wireless',
+                'uuid': uuid,
+                'type': '802-11-wireless'}, signature='sv'),
+            '802-11-wireless': dbus.Dictionary({
+                'ssid': dbus.ByteArray('The_SSID'.encode('UTF-8'))}, signature='sv')
+        }, signature='sa{sv}')
+        connectionPath = self.settings.AddConnection(settings)
+        self.assertEqual(self.settings.GetConnectionByUuid(uuid), connectionPath)
+
+        fakeuuid = '123123123213213'
+        with self.assertRaisesRegexp(dbus.exceptions.DBusException, ".*uuid.*%s$" % fakeuuid):
+            self.settings.GetConnectionByUuid(fakeuuid)
```

Comment:
Note you can use assertRaisesExp with `... as ex:` and then make assertions about the exception content, as you did before. That was actually useful :-)

Using a deprecated function in the tests is okay for now. If/once it goes away in Py 3, it can easily be aliased.